server/sv-main.lua

```diff
@@ -0,0 +1,191 @@
+    local playerId = source
+    local player = Framework.getPlayer(playerId)
+
+    if player and data then
+        if not IsReporter(playerId) then
+            Framework.showNotify(playerId, l('NOT_ALLOWED'), 'error')
+            return
+        end
+
+        local storyType = data.type and data.type:upper()
+
+        if storyType then
+            if STORIES_CACHE[storyType] then
+                local id = #STORIES_CACHE[storyType] + 1
+
+                STORIES_CACHE[storyType][id] = {
+                    title = data.title,
+                    story_type = storyType,
+                    image = data.image,
+                    jailed_player = '',
+                    jailed_time = 0,
+                    id = id,
+                    publisher = player.charName,
+                    body = data.body,
+                    date = data.date,
+                }
+
+                table.sort(STORIES_CACHE[storyType], function(a, b)
+                    return a.date > b.date
+                end)
+
+                Database.UpdateStory(data)
+
+                SyncStories(-1, STORIES_CACHE)
+
+                Framework.showNotify(playerId, l('STORY_UPDATED'), 'success')
+            else
+                print('futte-newspaper:server:updateStory: storyType is not valid')
+            end
+        else
+            print('futte-newspaper:server:updateStory: storyType is nil')
+        end
+    end
+end)
+
+RegisterNetEvent('futte-newspaper:server:publishStory', function(data)
+    local playerId = source
+    local player = Framework.getPlayer(playerId)
+
+    if player and data then
+        if not IsReporter(playerId) then
+            Framework.showNotify(playerId, l('NOT_ALLOWED'), 'error')
+            return
+        end
+
+        local storyType = data.type and data.type:upper()
+
+        if storyType then
+            if STORIES_CACHE[storyType] then
+                local id = #STORIES_CACHE[storyType] + 1
+
+                STORIES_CACHE[storyType][id] = {
+                    title = data.title,
+                    story_type = storyType,
+                    image = data.image,
+                    jailed_player = '',
+                    jailed_time = 0,
+                    id = id,
+                    publisher = player.charName,
+                    body = data.body,
+                    date = data.date,
+                }
+
+                table.sort(STORIES_CACHE[storyType], function(a, b)
+                    return a.date > b.date
+                end)
+
+                Database.PublishStory(data, player.charName)
+
+                SyncStories(-1, STORIES_CACHE)
+
+                Framework.showNotify(playerId, l('STORY_PUBLISHED'), 'success')
+            else
+                print('futte-newspaper:server:publishStory: storyType is not valid')
+            end
+        else
+            print('futte-newspaper:server:publishStory: storyType is nil')
+        end
```

Comment:
Some of it already has it, but could a couple of things be refactored to it has guardclauses instead? I think that would be more clean.

Reply:
Sure we can improve this part.